### PR TITLE
Implemented ratchets on midi clips, kit rows, unsynced rate, and stopped playback

### PR DESCRIPTION
**Pull request by soymonitus**
Before, you were only allowed to do ratchets on internal synth clips, and only if playback was engaged, and if arp rate was synced. Now it works in all other possible situations:
- Playing live notes on the audition pads with playback stopped
- Unsynced rate
- Kit rows
- Midi clips (which will output all the individual ratchet notes to your external synth!)

### CODE REVIEW

**Comment by soymonitus** [comment]
@seangoodvibes could you review this? Mark can't do it for a while

File: src/deluge/modulation/arpeggiator.cpp [hunk 1]
```diff
@@ -138,13 +133,13 @@ void ArpeggiatorForDrum::noteOff(ArpeggiatorSettings* settings, ArpReturnInstruc
 	}
 
 	arpNote.velocity = 0; // Means note is off
+	reset();
```
Comment (sapphire-arches) [inline src/deluge/modulation/arpeggiator.cpp:136]: What's the purpose of this reset?

File: src/deluge/modulation/arpeggiator.cpp [hunk 2]
```diff
@@ -647,57 +662,62 @@ void ArpeggiatorBase::render(ArpeggiatorSettings* settings, int32_t numSamples,
...
 	}
 
+	uint32_t gateThresholdSmall = gateThreshold >> 8;
+
 	if (isRatcheting) {
 		// shorten gate in case we are ratcheting (with the calculated number of ratchet notes)
 		gateThresholdSmall = gateThresholdSmall >> ratchetNotesMultiplier;
 	}
 
+	uint32_t maxGate = 16777216;
```
Comment (sapphire-arches) [inline src/deluge/modulation/arpeggiator.cpp:694]:
```suggestion
	uint32_t maxGate = 1 << 25;
```

make sure I got the shift amount right, but i think that's clearer.

Comment (soymonitus) [reply src/deluge/modulation/arpeggiator.cpp:694]: It is 1<< 24. If you can edit your proposal so i can confirm it

Comment (soymonitus) [reply src/deluge/modulation/arpeggiator.cpp]: Fixed

File: src/deluge/modulation/arpeggiator.cpp (hunk 1 above)
Comment (soymonitus) [reply src/deluge/modulation/arpeggiator.cpp]: Oh i see what you mean, that function does nothing here. Removed.

**Comment by soymonitus** [comment]
Changes ready @sapphire-arches

**Pull request merged**